File: 2170-count-number-of-maximum-bitwise-or-subsets/count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) {
        int n = nums.size();
        int maxi = 0;
        int curr = 0;
        int count = 0;
        for(int num:nums){
            curr = curr|num;
            maxi=max(maxi,curr);
        }
        n=1<<n;
        for(int i=0;i<n;i++){
            int curr = 0;
            for(int j=0;j<nums.size();j++){
                if(i&(1<<j)){
                    curr = curr|nums[j];
                }
            }
                    if(curr == maxi){
                        count++;
                    }
            
        }

        return count;            
    }
};

File: 2170-count-number-of-maximum-bitwise-or-subsets/count-number-of-maximum-bitwise-or-subsets.go
func countMaxOrSubsets(nums []int) int {
    maxi:=0
    for _,num:= range nums{
        maxi|=num
    }

    count:=0
    n:=len(nums)
    n=1<<n
    for i:=0;i<n;i++{
        curr:=0
        for j:=0;j<len(nums);j++{
            if (i&(1<<j))!=0{
                curr|=nums[j]
            }
        }
         
        if curr==maxi{
            count++
        }
    }
    return count
}
